client/verta/verta/runtime.py

```diff
@@ -0,0 +1,240 @@
+# -*- coding: utf-8 -*-
+"""Classes and functions to enable logging within a model's predict() function.
+.. versionadded:: 0.22.0
+
+"""
+
+import json
+import re
+import threading
+from typing import Any, Dict, Optional
+
+
+_THREAD = threading.local()
+_S3_REGEX = re.compile("[0-9a-zA-Z_-]+$")
+
+
+def _get_thread_logs() -> Dict[str, Any]:
+    """
+    Return the current thread-local logs or initialize an empty dict.
+    """
+    if not hasattr(_THREAD, 'logs'):
+        _set_thread_logs(dict())
+    return _THREAD.logs
+
+
+def _set_thread_logs(logs: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Set the thread-local logs, overwriting values for existing keys.
+    """
+    _THREAD.logs = logs
+    return _THREAD.logs
+
+
+def _delete_thread_logs() -> None:
+    """
+    Drop the `log` attribute from the thread local variable.
+    """
+    _THREAD.__delattr__('logs')
```

Comment:
Accessing a dunder directly like this isn't...great? As I proposed [in Slack](https://vertaai.slack.com/archives/C01RN3BFUMD/p1674685341003589?thread_ts=1674684085.583239&cid=C01RN3BFUMD): `del _THREAD.logs` would be perfectly idiomatic.

Alternatively if we're using `hasattr(_THREAD, 'logs')` everywhere, we might as well use `delattr(_THREAD, 'logs')` for consistency.